src/manage_sql_secinfo.c

```diff
@@ -2656,6 +2657,485 @@ insert_cve_from_entry (element_t entry, element_t last_modified,
+          " VALUES"
+          " ('%s', '%s', %i, %i,"
+          "  %0.1f, '%s', '%s', '%s')"
+          " ON CONFLICT (uuid) DO UPDATE"
+          " SET name = EXCLUDED.name,"
+          "     creation_time = EXCLUDED.creation_time,"
+          "     modification_time = EXCLUDED.modification_time,"
+          "     severity = EXCLUDED.severity,"
+          "     description = EXCLUDED.description,"
+          "     cvss_vector = EXCLUDED.cvss_vector,"
+          "     products = EXCLUDED.products"
+          " RETURNING scap2.cves.id;",
+          cve_id,
+          cve_id,
+          published_time,
+          modified_time,
+          score_dbl,
+          quoted_description,
+          vector,
+          "");
+
+  g_free (quoted_description);
+
+  cJSON *configurations_json;
+  cJSON *nodes_json;
+
+  configurations_json =
+       cJSON_GetObjectItemCaseSensitive(item, "configurations");
+  if (configurations_json == NULL)
+    {
+      g_warning("%s: configurations missing for %s.", __func__, cve_id);
+      return -1;
+    }
+  nodes_json =
+       cJSON_GetObjectItemCaseSensitive(configurations_json, "nodes");
+  if (nodes_json == NULL)
+    {
+      g_warning("%s: nodes missing for %s.", __func__, cve_id);
+      return -1;
+    }
+  load_nodes (0, cve_db_id, nodes_json);
+
+  return 0;
+}
+
+/**
+ * @brief Update CVE info from a single JSON feed file.
+ *
+ * @param[in]  cve_path          CVE json file path.
+ * @param[in]  hashed_cpes       Hashed CPEs.
+ *
+ * @return 0 success, -1 error.
+ */
+static int
+update_cve_json (const gchar *cve_path, GHashTable *hashed_cpes)
+{
+  cJSON *entry;
+  FILE *cve_file;
+  gchar *error_message = NULL;
+  gvm_json_pull_event_t event;
+  gvm_json_pull_parser_t parser;
+  gchar *full_path;
+  GStatBuf state;
+  int transaction_size = 0;
+  // int ret;
+
+  full_path = g_build_filename (GVM_SCAP_DATA_DIR, cve_path, NULL);
+
+  if (g_stat (full_path, &state))
+    {
+      g_warning ("%s: Failed to stat SCAP file: %s",
+                 __func__,
+                 strerror (errno));
+      g_free (full_path);
+      return -1;
+    }
+
+  g_info ("Updating %s", full_path);
+
+  cve_file = fopen (full_path, "r");
+  if (cve_file == NULL)
+    {
+      g_warning ("%s: Failed to open CVE file: %s",
+                 __func__,
+                 strerror (errno));
+      g_free (full_path);
+      return -1;
+    }
+
+  g_free (full_path);
+
+  gvm_json_pull_parser_init (&parser, cve_file);
+  gvm_json_pull_event_init (&event);
+  gvm_json_pull_parser_next (&parser, &event);
+
+  if (event.type == GVM_JSON_PULL_EVENT_OBJECT_START)
+    {
+      gboolean cve_items_found = FALSE;
+      while (!cve_items_found)
+        {
+          gvm_json_pull_parser_next (&parser, &event);
+	  gvm_json_path_elem_t *path_tail = g_queue_peek_tail (event.path);
+	  if (event.type == GVM_JSON_PULL_EVENT_ARRAY_START &&
+              strcmp (path_tail->key, "CVE_Items") == 0)
```

Comment:
Does it make sense to check `path_tail` for NULL here before accessing the key ?